build.rs

```diff
@@ -1,9 +1,19 @@
 use std::env;
 
 fn main() {
-    if env::var("HOST") != env::var("TARGET") {
-        println!("cargo:rustc-flags=-L libs -l jpeg -l png16 -l openjp2 -l jbig2dec -l bz2 -l z -l m");
+    if env::var("HOST") != env::var("TARGET") {   // assume cross-compiling for Kobo
         println!("cargo:rustc-env=PKG_CONFIG_ALLOW_CROSS=1");
-        println!("cargo:rustc-link-lib=dylib=stdc++");
+        println!("cargo:rustc-link-search=libs");
+    } else {                                      // assume compiling for host
+        println!("cargo:rustc-link-search=src/wrapper");
+        println!("cargo:rustc-link-lib=lcms2");
+        println!("cargo:rustc-link-lib=mupdf-third");
     }
+    println!("cargo:rustc-link-lib=z");
+    println!("cargo:rustc-link-lib=bz2");
+    println!("cargo:rustc-link-lib=jpeg");
+    println!("cargo:rustc-link-lib=png16");
+    println!("cargo:rustc-link-lib=openjp2");
+    println!("cargo:rustc-link-lib=jbig2dec");
+    println!("cargo:rustc-link-lib=dylib=stdc++");
```

Comment:
This need to stay inside the cross-compiling block or it will break the compilation on macOS.

Reply:
It is required on Linux. :-/
How does it break the compilation?

Reply:
It fails with: `ld: library not found for -lstdc++`.

Reply:
Ah yes, of course... It comes from #93: "Since the Kobo's Linaro toolchain is based around GCC, hardcode stdc++ and ignore Clang's libc++ possibility.", which is certainly true when building a Kobo binary, but the emulator's dependencies on recent enough OSX will link against clang's libc++.

What is the output of `llvm-config --cxxflags` on such machine?

Does changing the
```println!("cargo:rustc-link-lib=dylib=stdc++");```
line to
```println!("cargo:rustc-link-lib=dylib=c++");```
corrects the problem on OSX?

Reply:
> What is the output of `llvm-config --cxxflags` on such machine?

I don't have that binary on my system.

> Does changing the
> `println!("cargo:rustc-link-lib=dylib=stdc++");`
> line to
> `println!("cargo:rustc-link-lib=dylib=c++");`
> corrects the problem on OSX?

Yes it does.